### PR TITLE
Fix compilation for Windows

### PR DESCRIPTION
**Pull request by MicheleCancilla**
This PR solves some errors with Windows:
  1. `#include <sys/time.h>` has been wrapped using EDDL_WINDOWS macro.
  1. Some newer OpenMP directives have been omitted only for Windows, again using EDDL_WINDOWS macro.
  1. I updated the Jenkinsfile changing the base docker images. It should be useful for checking gpu-enabled tests of eddl.

### CODE REVIEW

File: src/net/net_api.cpp [hunk 1]
```diff
@@ -151,9 +152,13 @@ void Net::run_snets(void *(*F)(void *t))
   int rc;
   struct tdata td[100];
 
-  int comp=snets.size();
+  int comp = snets.size();
 
+#ifdef EDDL_WINDOWS
```
[the remark this answers is not in the document]
Comment (MicheleCancilla) [reply src/net/net_api.cpp:157]: For this simple usage _task_ (openmp 3+, while `taskloop` openmp 4.5+) construct should be equivalent to _parallel_. Things would be different in the case of recursuve functions where _task_ producer-consumer paradigm is preferred (performance or scalablity).

File: src/hardware/cpu/cpu_comparison.cpp [hunk 2]
```diff
@@ -155,7 +159,9 @@ bool cpu_allclose(Tensor *A, Tensor *B, float rtol, float atol, bool equal_nan){
                 if(first_idx < 0) { first_idx=i; }
 
             }
+#if defined(EDDL_LINUX) || defined(EDDL_UNIX) || defined(EDDL_APPLE)
```
[the remark this answers is not in the document]
Comment (MicheleCancilla) [reply src/hardware/cpu/cpu_comparison.cpp:162]: No, it requires OpenMP 4.0+

File: src/hardware/cpu/nn/cpu_losses.cpp [hunk 3]
```diff
@@ -31,7 +31,7 @@ float cpu_categorical_cross_entropy(Tensor* y_true, Tensor* y_pred){
     float eps = 10e-8;
 
     #pragma omp parallel for reduction(+:sum)
-    for (unsigned int bi = 0; bi<y_true->shape[0]; bi++) {  // Batches
```
[the remark this answers is not in the document]
Comment (MicheleCancilla) [reply src/hardware/cpu/nn/cpu_losses.cpp:34]: Yes,  indices can't be negative, but OpenMP 2 requires loop variables of signed types, so `int`, `long int` and so on. I put `long long` but `int` would have been better indeed.

File: src/CMakeLists.txt [hunk 4]
```diff
@@ -244,10 +244,10 @@ if(BUILD_OPENMP)
 #            endif()
 #        endif()
 
-        find_package(OpenMP 4.5)
-        if (OpenMP_CXX_FOUND)
+        find_package(OpenMP)
```
[the remark this answers is not in the document]
Comment (MicheleCancilla) [reply src/CMakeLists.txt:247]: Ok, I can enclose the find_package in `if` depending on the OS.

File: src/hardware/cpu/nn/cpu_losses.cpp (hunk 3 above)
[the remark this answers is not in the document]
Comment (salvacarrion) [reply src/hardware/cpu/nn/cpu_losses.cpp:34]: Good to know. Thanks!

File: src/CMakeLists.txt (hunk 4 above)
[the remark this answers is not in the document]
Comment (salvacarrion) [reply src/CMakeLists.txt]: That would be great. We can leave it later since now it is not a problem

**Pull request merged**